lib/node_modules/@stdlib/array/base/mskfilter-map/lib/assign.js

```diff
@@ -0,0 +1,245 @@
+var arraylike2object = require( '@stdlib/array/base/arraylike2object' );
+var reinterpret = require( '@stdlib/strided/base/reinterpret-complex' );
+
+
+// FUNCTIONS //
+
+/**
+* Applies a mask to an indexed array & apply mapping function and assigns unmasked values to elements in an indexed output array.
+*
+* @private
+* @param {Collection} x - input array
+* @param {IntegerArray} mask - mask array
+* @param {Collection} out - output array
+* @param {integer} stride - output array stride
+* @param {NonNegativeInteger} offset - output array offset
+* @param {Function} clbk - function to apply
+* @param {*} [thisArg] - function execution context
+* @returns {Collection} output array
+*
+* @example
+* var x = [ 1, 2, 3, 4 ];
+* var mask = [ 0, 1, 0, 1 ];
+*
+* var out = [ 0, 0, 0, 0 ];
+*
+* function mappingFunction( x ) {
+*     return x * 10;
+* }
+*
+* var arr = indexed( x, mask, out, 1, 0, mappingFunction );
+* // returns [ 20, 40, 0, 0 ]
+*/
+function indexed( x, mask, out, stride, offset, clbk, thisArg ) {
+	var io;
+	var i;
+
+	io = offset;
+	for ( i = 0; i < x.length; i++ ) {
+		if ( mask[ i ] ) {
+			out[ io ] = clbk.call( thisArg, x[ i ] );
+			io += stride;
+		}
+	}
+	return out;
+}
+
+/**
+* Applies a mask to an accessor array & apply mapping function and assigns unmasked values to elements in an accessor output array.
+*
+* @private
+* @param {Object} x - input array object
+* @param {Object} mask - mask array object
+* @param {Object} out - output array object
+* @param {integer} stride - output array stride
+* @param {NonNegativeInteger} offset - output array offset
+* @param {Function} clbk - function to apply
+* @param {*} [thisArg] - function execution context
+* @returns {Collection} output array
+*
+* @example
+* var toAccessorArray = require( '@stdlib/array/base/to-accessor-array' );
+* var arraylike2object = require( '@stdlib/array/base/arraylike2object' );
+*
+* var x = toAccessorArray( [ 1, 2, 3, 4 ] );
+* var mask = toAccessorArray( [ 0, 1, 0, 1 ] );
+*
+* function mappingFunction( x ) {
+*     return x * 10;
+* }
+*
+* var out = toAccessorArray( [ 0, 0, 0, 0 ] );
+* var arr = accessors( arraylike2object( x ), arraylike2object( mask ), arraylike2object( out ), 1, 0, mappingFunction );
+*
+* var v = arr.get( 0 );
+* // returns 20
+*
+* v = arr.get( 1 );
+* // returns 40
+*/
+function accessors( x, mask, out, stride, offset, clbk, thisArg ) {
+	var xdata;
+	var mdata;
+	var odata;
+	var xget;
+	var mget;
+	var oset;
+	var io;
+	var i;
+
+	xdata = x.data;
+	mdata = mask.data;
+	odata = out.data;
+
+	xget = x.accessors[ 0 ];
+	mget = mask.accessors[ 0 ];
+	oset = out.accessors[ 1 ];
+
+	io = offset;
+	for ( i = 0; i < xdata.length; i++ ) {
+		if ( mget( mdata, i ) ) {
+			oset( odata, io, clbk.call( thisArg, xget( xdata, i ) ) );
+			io += stride;
+		}
+	}
+	return odata;
+}
+
+/**
+* Applies a mask to a complex array & apply mapping function and assigns unmasked values to elements in a complex output array.
```

Comment:
```suggestion
* Applies a mask and mapping function to a complex array and assigns results to elements in an output array.
```

The output array need not be complex.